Simplifies the cln plugin option parsing

sr-gi opened the pull request:
Options include helper functions to convert them to their proper type after
`cln-plugin=0.1.2`. Use that to reduce the option parsing boilerplate.

Also move all names, descriptions and default values for options, rpc_methods and hooks
to a new file.

mariocynicys commented on watchtower-plugin/src/main.rs:
```diff
@@ -65,14 +67,23 @@ async fn register(
-        tower_net_addr = format!("http://{}", tower_net_addr)
-    }
+    let tower_net_addr = {
+        if !host.starts_with("http") {
+            host = format!("http://{}", host)
+        }
```
nit: `host` isn't quite the name for an HTTP url.
Also let's check `.starts_with("http://")` because a host name might actually start with `http` :laughing:.

mariocynicys reviewed (approved):
A couple of nits, all fine otherwise.

mariocynicys commented:
Ok now it makes sense why the PR's name is not very inclusive.
I think you included changes from #160 in here by mistake in [`4c9141f`](https://github.com/talaia-labs/rust-teos/pull/170/commits/4c9141fd1042a57d8194945112d732326337b5f5).

sr-gi commented:
Oh yeah, this is not pulling from master giving `cln-plugin=0.1.2` is bumped in #160, and this PR builds on top of that.

This will need rebasing once #160 gets merged, but given it was double at this stage and didn't require much more I went ahead and fixed it.

sr-gi commented:
> Rebased and amended this so it pulls from master.
> 
> I've also moved a bit more data to `constants.rs` to reduce the hardcoded stuff in `main.rs`.

Looks like I've broken the plugin by doing this 😅, not exactly sure why though.

sr-gi commented:
> > Rebased and amended this so it pulls from master.
> > I've also moved a bit more data to `constants.rs` to reduce the hardcoded stuff in `main.rs`.
> 
> Looks like I've broken the plugin by doing this 😅, not exactly sure why though.

Fixed it, `auto-retry-delay` default value was beyond `u16::MAX`. Looks like this was not caught in https://github.com/talaia-labs/rust-teos/commit/07caee2dbda894db462768290af2c922f1e30124 because the default was set in two different places on the codebase:

https://github.com/talaia-labs/rust-teos/blob/07caee2dbda894db462768290af2c922f1e30124/watchtower-plugin/src/main.rs#L544-L547

https://github.com/talaia-labs/rust-teos/blob/07caee2dbda894db462768290af2c922f1e30124/watchtower-plugin/src/main.rs#L625-L629

mariocynicys commented:
One of the failures though wasn't in the e2e tests: https://github.com/talaia-labs/rust-teos/actions/runs/3884258978/jobs/6626583255

This panicked while checking if the retrier is idle. Guess the sleep time is too tight?

sr-gi commented:
> @sr-gi One of the failures though wasn't in the e2e tests: https://github.com/talaia-labs/rust-teos/actions/runs/3884258978/jobs/6626583255
> 
> This panicked while checking if the retrier is idle. Guess the sleep time is too tight?

That's my guess, unfortunately, I haven't been able to reproduce it locally
